### ISSUE TITLE
Northstar auto-installation

### ISSUE THREAD

**Issue by Alystrasz** [0neGal/viper#25]
When first launching Viper, the user say where is located his Titanfall game.
Once done, he can click the "Northstar" button to launch the game, which will fail, because Northstar has not been installed on his folder.

I think we should replace "Northstar" button text with "Install" (if `NorthstarLauncher.exe` is not found for instance) and install ns on first click, instead of trying to run an executable we know is not there.

**Comment by 0neGal** [comment]
> I think we should replace "Northstar" button text with "Install"

I think a better approach would be to disable the launch button for Northstar, and set the text of the "Update" button to "Install". Perhaps even "Install Northstar" althought that might be very long.

**Comment by Alystrasz** [comment]
Yeah my bad, I though of it from my new UI branch perspective, which features a big "PLAY" button.

![Capture d’écran 2022-01-04 184333](https://user-images.githubusercontent.com/11993538/148101342-283065d0-8a5d-4610-ad0c-2d0e09860b27.png)

**Comment by 0neGal** [comment]
In that case yes, your suggestion would make sense, we should probably also set the northstar version to "Not Installed" on the frontend.

**Comment by Alystrasz** [comment]
Also, I think we should check every X minutes if a Northstar update is available.
I don't think users will click the "update" button each time they launch Northstar.

**Comment by Alystrasz** [comment]
@0neGal what's your opinion on this?

**Comment by 0neGal** [comment]
Sorry I've been quite busy with family business and what not (hence why I haven't looked at your PR, will do later today)

>>Also, I think we should check every X minutes if a Northstar update is available. I don't think users will click the "update" button each time they launch Northstar.
> @0neGal what's your opinion on this?

That is a good idea. Also maybe make a background service that keeps Northstar up to date...?

**Comment by Alystrasz** [comment]
Yeah we can check every 10 minutes if there's an update available for instance.
